test/integration.js

```diff
@@ -0,0 +1,132 @@
+/* globals beforeAll: false, afterAll: false, describe: false, it: false, expect: false, tester: false, storage: false */
+import sinon from 'sinon'
+
+let searcher = tester.getApplicationModule('searcher')
+
+function timeout(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+function processor(doc) {
+  doc.lower_name = doc.name.toLowerCase()
+  return doc
+  
+}
+processor = sinon.spy(processor)
+
+describe('Integration', () => {
+  describe("searchable", () => {
+    it("registers model", async () => {
+      await searcher.searchable('test-model')
+      
+      expect(searcher._instance.modelConfig).toHaveProperty('test-model', {
+        fields: ['name', 'title', 'description'],
+        model: 'test-model',
+        searchdocument: 'searchdocument'
+      })
+
+      expect(searcher._instance._searchDocuments).toHaveProperty('', 'searchdocument')
+    })
+    it("registers model with processor", async () => {
+      await searcher.searchable('test-model2', {processor})
+      
+      expect(searcher._instance.modelConfig['test-model2']).toHaveProperty('processor')
+
+    })
+
+  })
+
+  describe("documents", () => {
+    it("creates document", async () => {
+      let model = await storage.getModel('test-model')
+      let val = {name: 'Test Model'}
+      await model.create(val)
+      // timeout and refresh for model events not awaited
+      await timeout(200)
+      await tester.searcherRefresh()
+
+      let count = await searcher.count('test-model', 'Test')
+      expect(count).toEqual(1)
+      let res = await searcher.search('test-model', 'Test')
+      expect(res.total).toEqual(1)
+      expect(res[0].model).toEqual('test-model')
+      expect(res[0].name).toEqual(val.name)
+    })
+    it("creates document with processing", async () => {
+      let model = await storage.getModel('test-model2')
+      let val = {name: 'Test Model'}
+      await model.create(val)
+      // timeout and refresh for model events not awaited
+      await timeout(200)
+      await tester.searcherRefresh()
+
+      expect(processor.callCount).toEqual(1)
+      let res = await searcher.search('test-model2', 'Test')
+      expect(res.total).toEqual(1)
+      expect(res[0].model).toEqual('test-model2')
+      expect(res[0].lower_name).toEqual(val.name.toLowerCase())
+    })
+
+    it("updates document", async () => {
+      let model = await storage.getModel('test-model')
+      let m = await model.findOne({name: 'Test Model'})
+      m.name = 'Untest Model'
+      await m.save()
+      // timeout and refresh for model events not awaited
+      await timeout(200)
+      await tester.searcherRefresh()
+      
+      let res = await searcher.search('test-model', 'Model')
```

Comment:
Aren't there two things to check here – the new searcher index term is present, and the old term is gone. (Update name to "Modified Model" and check that "Test" is gone and "Modified" is present?)

Reply:
Agreed, will add the negative test.